File: backend/internal/entity/quiz.go
// WIP file might not work properly.

package entity

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

// Quiz struct
type Quiz struct {
	Id        primitive.ObjectID `json:"id" bson:"_id"`
	Name      string             `json:"name"`
	Questions []QuizQuestion     `json:"questions"`
}

// QuizQuestion struct
type QuizQuestion struct {
	Id      string       `json:"id"`
	Name    string       `json:"name"`
	Choices []QuizChoice `json:"choices"`
}

// QuizChoice struct
type QuizChoice struct {
	Id      string `json:"id"`
	Name    string `json:"name"`
	Correct bool   `json:"correct"`
}

File: frontend/src/model/quiz.ts
// Objective: Define the interfaces for the quiz model.


export interface Quiz {
    id: string;
    name: string;
    questions: QuizQuestion[];
}

export interface Player {
    id: string;
    name: string;
}

export interface QuizQuestion {
    name: any;
    id: string;
    question: string;
    choices: QuizChoice[];
}

export interface QuizChoice {
    [x: string]: any;
    text: any;
    id: string;
    choice: string;
    correct: boolean;
}
